projects/js-packages/charts/src/components/legend/use-chart-legend-data.ts

```diff
@@ -124,7 +135,11 @@ function processPointData(
 		const baseItem = createBaseLegendItem(
 			point.label,
 			formatPointValue( point, showValues ),
-			theme.colors[ index % theme.colors.length ]
+			( point as DataPointPercentage ).color ?? theme.colors[ index % theme.colors.length ],
```

Comment:
I don't fully understand the reason for the type casting here 🤔

Reply:
Type is `DataPointDate | DataPointPercentage`, and `DataPointDate` doesn't have a color property, so TS complains if you try to reference `point.color`